### PR TITLE
add bash completion

### PR DESCRIPTION
**Pull request by 1ace**

### CODE REVIEW

File: shell-completion/bash/bear [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      -o|--cdb)
+        _filedir
+        COMPREPLY+=($(compgen -W 'compile_commands.json' -- "$cur"))
+        return
+        ;;
+      --use-cc|--use-c++)
+        _command_offset $COMP_CWORD
+        return
+        ;;
+      -l|--libear)
```
Comment (rizsotto) [inline shell-completion/bash/bear]: this is more like developer option. (supposed to used in tests)

**Comment by rizsotto** [comment]
Thanks Eric for the PR. I'm wondering would people use it? And does how does this offers the build command part? Does this offers the `--` to separate the Bear arguments from the build command?

**Comment by 1ace** [comment]
Sorry for completely forgetting to reply and follow up on this PR :speak_no_evil:

> I'm wondering would people use it?

"Build it and they will come" :stuck_out_tongue_winking_eye:
Seriously though, I think shell completion is something most people find useful, so yeah I think people will use it.

> And does how does this offers the build command part?

It does that using the `_command_offset` function, which basically means "the next word is an independent command".

> Does this offers the `--` to separate the Bear arguments from the build command?

It didn't, but I've just added that :+1:

File: shell-completion/bash/bear (hunk 1 above)
Comment (1ace) [reply shell-completion/bash/bear]: I removed the "advanced options" :+1:

**Comment by rizsotto** [comment]
Thanks Eric!

**Comment by 1ace** [comment]
Don't hesitate to tag me on any PR that you merge that adds arguments, I'm happy to write the bash completion code if you're not sure how to do it yourself :+1:

**Pull request merged**